src/libraries/System.Net.Http/src/System/Net/Http/Headers/ObjectCollection.cs

```diff
@@ -120,7 +121,7 @@ public void CopyTo(T[] array, int arrayIndex)
 
         public bool Remove(T item)
         {
-            if (ReferenceEquals(_items, item))
+            if (_items is T o && o.Equals(item))
```

Comment:
If _items is a T but it's not equal to the item, there's no use in then subsequently checking whether _items is a T[].  So this should instead be structured like:
```C#
if (_items is T o)
{
    if (o.Equals(item))
    {
        _items = null;
        _size = 0;
        return true;
    }
}
else if (_items is T[] items)
{
    ...
}

return false;
```